server/tests/test_api.py

```diff
@@ -1,40 +1,146 @@
-import os
-import tempfile
+"""Test the api."""
+from typing import Any, Generator, cast
+from unittest.mock import Mock
 
 import pytest
-from src.api import get_app
+from flask.testing import FlaskClient
+from pytest_mock import MockerFixture
+from src.api import IsLoggedInResponse, LoginResponse, app
+from src.model.user import User
+from src.utils.error import AppError, ErrorCode
+
+CSFR_HEADERS = {"X-Requested-With": "XmlHttpRequest"}
 
 
 @pytest.fixture
-def client():
+def client() -> Generator[FlaskClient, None, None]:
     """Get client api app for testing."""
-    db_fd, db_path = tempfile.mkstemp()
-    app = get_app()
-    # app = create_app({'TESTING': True, 'DATABASE': db_path})
-
+    app.secret_key = "test_key"
     with app.test_client() as client:
         yield client
 
-    os.close(db_fd)
-    os.unlink(db_path)
-
 
-def test_test(client):
+def test_test_route(client: FlaskClient) -> None:
     """Test that test endpoint is working."""
     res = client.get("/test")
     assert res.status_code == 200
-    assert b"The test is successful" in res.data
+    assert b"test is successful" in res.data
 
 
-def test_hello(client):
+def test_non_existent_route(client: Any) -> None:
```

Comment:
Why is this typed as `Any` but others as `FlaskClient`?

Reply:
Cause it was the first one i wrote before i knew what i was doing lol.